README.rst
Python Client for Virtual Private Cloud API
===========================================

|stable| |pypi| |versions|

`Virtual Private Cloud API`_: 

- `Client Library Documentation`_
- `Product Documentation`_

.. |stable| image:: https://img.shields.io/badge/support-stable-gold.svg
   :target: https://github.com/googleapis/google-cloud-python/blob/main/README.rst#stability-levels
.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-vpc-access.svg
   :target: https://pypi.org/project/google-cloud-vpc-access/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-vpc-access.svg
   :target: https://pypi.org/project/google-cloud-vpc-access/
.. _Virtual Private Cloud API: https://cloud.google.com/vpc/
.. _Client Library Documentation: https://cloud.google.com/python/docs/reference/vpcaccess/latest
.. _Product Documentation:  https://cloud.google.com/vpc/

Quick Start
-----------

In order to use this library, you first need to go through the following steps:

1. `Select or create a Cloud Platform project.`_
2. `Enable billing for your project.`_
3. `Enable the Virtual Private Cloud API.`_
4. `Setup Authentication.`_

.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
.. _Enable the Virtual Private Cloud API.:  https://cloud.google.com/vpc/
.. _Setup Authentication.: https://googleapis.dev/python/google-api-core/latest/auth.html

Installation
~~~~~~~~~~~~

Install this library in a `virtualenv`_ using pip. `virtualenv`_ is a tool to
create isolated Python environments. The basic problem it addresses is one of
dependencies and versions, and indirectly permissions.

With `virtualenv`_, it's possible to install this library without needing system
install permissions, and without clashing with the installed system
dependencies.

.. _`virtualenv`: https://virtualenv.pypa.io/en/latest/


Code samples and snippets
~~~~~~~~~~~~~~~~~~~~~~~~~

Code samples and snippets live in the `samples/` folder.


Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Our client libraries are compatible with all current [active](https://devguide.python.org/devcycle/#in-development-main-branch) and [maintenance](https://devguide.python.org/devcycle/#maintenance-branches) versions of
Python.

Python >= 3.7

Unsupported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^^
Python <= 3.6

If you are using an [end-of-life](https://devguide.python.org/devcycle/#end-of-life-branches)
version of Python, we recommend that you update as soon as possible to an actively supported version.


Mac/Linux
^^^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    source <your-env>/bin/activate
    <your-env>/bin/pip install google-cloud-vpc-access


Windows
^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    <your-env>\Scripts\activate
    <your-env>\Scripts\pip.exe install google-cloud-vpc-access

Next Steps
~~~~~~~~~~

-  Read the `Client Library Documentation`_ for Virtual Private Cloud API
   to see other available methods on the client.
-  Read the `Virtual Private Cloud API Product documentation`_ to learn
   more about the product and see How-to Guides.
-  View this `README`_ to see the full list of Cloud
   APIs that we cover.

.. _Virtual Private Cloud API Product documentation:  https://cloud.google.com/vpc/
.. _README: https://github.com/googleapis/google-cloud-python/blob/main/README.rst


CHANGELOG.md
# Changelog

### [1.0.1](https://www.github.com/googleapis/python-vpc-access/compare/v1.0.0...v1.0.1) (2021-09-24)


### Bug Fixes

* add 'dict' annotation type to 'request' ([be61638](https://www.github.com/googleapis/python-vpc-access/commit/be616386c0a9db4c49f8f319498411ab969542a3))


### Documentation

* migrate default branch from master to main ([#34](https://www.github.com/googleapis/python-vpc-access/issues/34)) ([7f5d990](https://www.github.com/googleapis/python-vpc-access/commit/7f5d99035c709cc67ed7ef61629936147bef0bac))

## [1.0.0](https://www.github.com/googleapis/python-vpc-access/compare/v0.2.1...v1.0.0) (2021-08-03)


### Features

* bump release level to production/stable ([#29](https://www.github.com/googleapis/python-vpc-access/issues/29)) ([6e8f2b1](https://www.github.com/googleapis/python-vpc-access/commit/6e8f2b1a5abd697da122854f8fee4c8d3cb00383))

### [0.2.1](https://www.github.com/googleapis/python-vpc-access/compare/v0.2.0...v0.2.1) (2021-07-26)


### Bug Fixes

* **deps:** pin 'google-{api,cloud}-core', 'google-auth' to allow 2.x versions ([#20](https://www.github.com/googleapis/python-vpc-access/issues/20)) ([46a4eaf](https://www.github.com/googleapis/python-vpc-access/commit/46a4eaf7814d69edb7b5ecb1767805088e3e82f9))
* enable self signed jwt for grpc ([#26](https://www.github.com/googleapis/python-vpc-access/issues/26)) ([aca8358](https://www.github.com/googleapis/python-vpc-access/commit/aca8358bf75e76a49508688507aba3d73ec8d95c))


### Documentation

* add Samples section to CONTRIBUTING.rst ([#21](https://www.github.com/googleapis/python-vpc-access/issues/21)) ([d1fc404](https://www.github.com/googleapis/python-vpc-access/commit/d1fc404fd34d69b70c925bf3af2a022c116a5a11))


### Miscellaneous Chores

* release 0.2.1 ([#25](https://www.github.com/googleapis/python-vpc-access/issues/25)) ([8ded00a](https://www.github.com/googleapis/python-vpc-access/commit/8ded00a39a9f151376e1060b617805222aef78c9))

## [0.2.0](https://www.github.com/googleapis/python-vpc-access/compare/v0.1.0...v0.2.0) (2021-07-14)


### Features

* add always_use_jwt_access ([#11](https://www.github.com/googleapis/python-vpc-access/issues/11)) ([6f1f049](https://www.github.com/googleapis/python-vpc-access/commit/6f1f0499f661625e77c71543f9b70f60b4478338))


### Bug Fixes

* disable always_use_jwt_access ([#15](https://www.github.com/googleapis/python-vpc-access/issues/15)) ([25a9da1](https://www.github.com/googleapis/python-vpc-access/commit/25a9da1e9b7761632befd3b0e7646f7e45f2ebc2))


### Documentation

* omit mention of Python 2.7 in 'CONTRIBUTING.rst' ([#1127](https://www.github.com/googleapis/python-vpc-access/issues/1127)) ([#6](https://www.github.com/googleapis/python-vpc-access/issues/6)) ([36a763a](https://www.github.com/googleapis/python-vpc-access/commit/36a763acbccf7641efc4d57fcb7ebddf3322d66a))

## 0.1.0 (2021-06-14)


### Features

* generate v1 ([2f99f6f](https://www.github.com/googleapis/python-vpc-access/commit/2f99f6f08c23ac14df17deef6c1d131e396a8e2c))


owl-bot-staging/v1/noxfile.py
# -*- coding: utf-8 -*-
# Copyright 2020 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
import os
import pathlib
import shutil
import subprocess
import sys


import nox  # type: ignore

CURRENT_DIRECTORY = pathlib.Path(__file__).parent.absolute()

LOWER_BOUND_CONSTRAINTS_FILE = CURRENT_DIRECTORY / "constraints.txt"
PACKAGE_NAME = subprocess.check_output([sys.executable, "setup.py", "--name"], encoding="utf-8")


nox.sessions = [
    "unit",
    "cover",
    "mypy",
    "check_lower_bounds"
    # exclude update_lower_bounds from default
    "docs",
]

@nox.session(python=['3.6', '3.7', '3.8', '3.9'])
def unit(session):
    """Run the unit test suite."""

    session.install('coverage', 'pytest', 'pytest-cov', 'asyncmock', 'pytest-asyncio')
    session.install('-e', '.')

    session.run(
        'py.test',
        '--quiet',
        '--cov=google/cloud/vpcaccess_v1/',
        '--cov-config=.coveragerc',
        '--cov-report=term',
        '--cov-report=html',
        os.path.join('tests', 'unit', ''.join(session.posargs))
    )


@nox.session(python='3.7')
def cover(session):
    """Run the final coverage report.
    This outputs the coverage report aggregating coverage from the unit
    test runs (not system test runs), and then erases coverage data.
    """
    session.install("coverage", "pytest-cov")
    session.run("coverage", "report", "--show-missing", "--fail-under=100")

    session.run("coverage", "erase")


@nox.session(python=['3.6', '3.7'])
def mypy(session):
    """Run the type checker."""
    session.install('mypy', 'types-pkg_resources')
    session.install('.')
    session.run(
        'mypy',
        '--explicit-package-bases',
        'google',
    )


@nox.session
def update_lower_bounds(session):
    """Update lower bounds in constraints.txt to match setup.py"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'update',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )


@nox.session
def check_lower_bounds(session):
    """Check lower bounds in setup.py are reflected in constraints file"""
    session.install('google-cloud-testutils')
    session.install('.')

    session.run(
        'lower-bound-checker',
        'check',
        '--package-name',
        PACKAGE_NAME,
        '--constraints-file',
        str(LOWER_BOUND_CONSTRAINTS_FILE),
    )

@nox.session(python='3.6')
def docs(session):
    """Build the docs for this library."""

    session.install("-e", ".")
    session.install("sphinx<3.0.0", "alabaster", "recommonmark")

    shutil.rmtree(os.path.join("docs", "_build"), ignore_errors=True)
    session.run(
        "sphinx-build",
        "-W",  # warnings as errors
        "-T",  # show full traceback on exception
        "-N",  # no colors
        "-b",
        "html",
        "-d",
        os.path.join("docs", "_build", "doctrees", ""),
        os.path.join("docs", ""),
        os.path.join("docs", "_build", "html", ""),
    )
